v1/ast/term.go

```diff
@@ -1644,6 +1644,10 @@ func (s *set) Find(path Ref) (Value, error) {
 
 // Diff returns elements in s that are not in other.
 func (s *set) Diff(other Set) Set {
+	if s.Compare(other) == 0 {
+		return InternedEmptySet.Value.(Set)
+	}
```

Comment:
There's some risk that a future built-in might start modifying the returned set. At the very least we should add a warning to the description. (Unless this cuts Regal eval time in half, I might even want to opt out of this change 😅)

Reply:
Well, the returned `Value` is still a mutable `Set`, so a user might still be putting values into that shared not-so-empty-anymore set, right?

Reply:
Yeah, I guess I can back out of this.

Reply:
Done, now returning `NewSet()` instead, so should still be somewhat cheaper than before.